### DISCUSSION TITLE
Additional one to one relation between models.

### DISCUSSION THREAD

**Discussion by alucardu** [prisma/prisma#18157] [Q&A] [answered]
I have two models:

```gql
model User {
  id        Int       @id @default(autoincrement())
  name      String
  messages  Message[] @relation(name: "userMessage")
  messageId Int?
  session   Session?  @relation(fields: [sessionId], references: [id], onDelete: Cascade)
  sessionId Int?

  @@unique([sessionId, name])
}

model Session {
  id       Int       @id @default(autoincrement())
  name     String
  code     String    @unique
  host     User
  players  User[]
  messages Message[]
}
```

A User and a Session model. This is working as intended. A user can have 1 session, a session can have many users and in a session the name has to be unique.

Now I'm trying to add an additional relation. A session should also have a `host`. A host in this case is the user that creates the session. (for example, the host can kick people from a session, or delete a session etc.)

But I'm having some trouble setting up the schema correctly:

```gql
model User {
  id        Int       @id @default(autoincrement())
  name      String
  messages  Message[] @relation(name: "userMessage")
  messageId Int?
  session   Session?  @relation(fields: [sessionId], references: [id], onDelete: Cascade)
  sessionId Int?
  host      Session?  @relation(name: "sessionHost")

  @@unique([sessionId, name])
}

model Session {
  id       Int       @id @default(autoincrement())
  name     String
  code     String    @unique
  players  User[]
  messages Message[]
  host     User      @relation(name: "sessionHost", fields: [userId], references: [id])
  userId   Int       @unique
}
```

When I view the app in the browser I get the error:

> Property 'userId' is missing in type 'import("C:/Users/alucardu/Documents/projects/ti4-utomation/src/types/sessionTypes").Session' but required in type 'import("C:/Users/alucardu/Documents/projects/ti4-automation/node_modules/.prisma/client/index").Session'.

**Comment by nurul3101** [comment]
Hey @alucardu 👋 

Do you get any error if you invoke `npx prisma format` command?
Alternatively, you can install Prisma VS Code Extension to highlight any issues in the schema file.

**Comment by alucardu** [comment, marked as answer]
@nurul3101 Apologies, I jumped the gun a little bit on this question. 

This was an TypeScript issue.  I have a session type defined:

```typescript
  type Session {
    id: ID
    code: String
    name: String
    players: [User]
  }
  ```
but it was missing the `userId: number;` property. After adding that my client issue was resolved.

I'm not sure how or why changing a field in the Prisma Schema would affect my client but it did.

**Reply by nurul3101** [reply]
Ah, thanks for getting back and letting us know! 👍
